### PR TITLE
CHK-1209-call-post-calculate-fee

### PR DESCRIPTION
**Pull request by Fernando-Granato**
#### List of Changes
Modified the call to the payment-methods for the GET of the psps during the phase of authorization of the payment.

#### Motivation and Context
Invoked the new primitive of the payment-methods POST fee/calculate in phase of authorization of a payment.
Integrated in docker-compose GEC’s mock server json.

#### How Has This Been Tested?
The tests were performed in local.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: Dockerfile.gec-mock [hunk 1]
```diff
@@ -0,0 +1,17 @@
+### 1. Cloning stage
+FROM alpine/git:latest AS clone
+WORKDIR /workspace/payment-instrument
+RUN git clone https://github.com/pagopa/pagopa-ecommerce-payment-instruments-service.git
+WORKDIR /workspace/payment-instrument/pagopa-ecommerce-payment-instruments-service
```
Comment (Fernando-Granato) [inline Dockerfile.gec-mock:5]: 👀

File: Dockerfile.payment-instruments [hunk 2]
```diff
@@ -2,6 +2,8 @@
 FROM alpine/git:latest AS clone
 WORKDIR /workspace/payment-instrument
 RUN git clone https://github.com/pagopa/pagopa-ecommerce-payment-instruments-service
+WORKDIR /workspace/payment-instrument/pagopa-ecommerce-payment-instruments-service
+RUN git checkout CHK-905-afm
```
Comment (Fernando-Granato) [inline Dockerfile.payment-instruments:6]: 👀

File: pom.xml [hunk 3]
```diff
@@ -263,7 +263,7 @@
                             <goal>generate</goal>
                         </goals>
                         <configuration>
-                            <inputSpec>https://raw.githubusercontent.com/pagopa/pagopa-ecommerce-payment-instruments-service/main/api-spec/api.yaml</inputSpec>
+                            <inputSpec>https://raw.githubusercontent.com/pagopa/pagopa-ecommerce-payment-instruments-service/CHK-905-afm/api-spec/api.yaml</inputSpec>
```
Comment (Fernando-Granato) [inline pom.xml:266]: 👀

File: src/main/java/it/pagopa/transactions/services/TransactionsService.java [hunk 4]
```diff
@@ -184,21 +183,45 @@ public Mono<RequestAuthorizationResponseDto> requestTransactionAuthorization(
...
+                                                    .paymentMethodId(
+                                                            requestAuthorizationRequestDto.getPaymentInstrumentId()
+                                                    )
+                                                    .touchpoint(transaction.getClientId().toString())
+                                                    .bin(
+                                                            requestAuthorizationRequestDto
+                                                                    .getDetails()instanceof CardAuthRequestDetailsDto cardData
+                                                                            ? cardData.getPan().substring(0, 8)
+                                                                            : null
+                                                    )
```
Comment (ciuffagianluca) [inline src/main/java/it/pagopa/transactions/services/TransactionsService.java:198]: Maybe an util method like `extractBin()` is better to use.

```
private Optional<string> extractBin(RequestAuthorizationRequestDetailsDto details) {
  //logic to extract bin
}

.bin(
                                                            extractBin().orElse("")
                                                    )
```

Comment (Fernando-Granato) [reply src/main/java/it/pagopa/transactions/services/TransactionsService.java:198]: orElse(""))must return a null value to create a GEC request what do you think ?

File: Dockerfile.gec-mock (hunk 1 above)
Comment (infantesimone) [reply Dockerfile.gec-mock:5]: the entire file :-)

File: pom.xml (hunk 3 above)
Comment (MarcoAGreco) [reply pom.xml:266]: Can we restore the endpoint to the main branch?

File: src/main/java/it/pagopa/transactions/client/EcommercePaymentInstrumentsClient.java [hunk 5]
```diff
@@ -1,27 +1,42 @@
...
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
 import reactor.core.publisher.Mono;
 
 @Component
+@Slf4j
 public class EcommercePaymentInstrumentsClient {
```
Comment (MarcoAGreco) [inline src/main/java/it/pagopa/transactions/client/EcommercePaymentInstrumentsClient.java:15]: Can we refactor this class renaming it to `EcommercePaymentMethodsClient`?

**Review by MarcoAGreco — DISMISSED** [review]
Nice! 🚀

**Review by ciuffagianluca — APPROVED** [review]
LGTM!! 🥇

**Pull request merged**